src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -223,6 +256,10 @@ Task AbandonAsyncOnCancellation(BrokeredMessage message)
                 logger.Error("The receiver failed to stop with in the time allowed (30s)");
             }
 
+            // drain the last messages before signaling to stop batched completion task
+            while (!locksIdsToComplete.IsEmpty) {}
+            lockIdCompletionCts.Cancel();
```

Comment:
I would swap these lines, and just call CompleteBatchAsync with the remaining lockids in the list instead of spinning this thread in a while loop